### ISSUE TITLE
Priority of integer literals is wrong

### ISSUE THREAD

**Issue by konnov** [informalsystems/quint#578]
As @thpani noticed, the following expression breaks in REPL:

```
 -1.to(10)
static analysis error: <input>:1:1 - error: Couldn't unify int and set
Trying to unify int and Set[int]
Trying to unify (int) => int and (Set[int]) => t9
```

Apparently, I have broken priorities in the parser once again.

**Comment by konnov** [comment]
Oh, the parser is only parsing positive integers. It explains the issue :)

**Comment by konnov** [comment]
This happened to be a much more complex question that I thought. We could fix the grammar to deal with `-1.to(10)`. However, it would be impossible to fix the grammar to work in the general case `-x.to(y)`, as the precedence of `.` is higher than the precedence of the unary minus, not only in Quint, but in many languages.

If we choose to define `-1` as an integer literal, not as `-(1)`, which is the current way it is recognized by the parser, then it would conflict with the precedence of `x^y`. In Python and TLA+, `-2 ** 4` and `-2^4` return `-16`, not `16`. This shows that Python and TLA+ do not treat `-2` as an integer literal, but parse the whole expression as `-(2 ** 4)`. It also shows that the power operator has higher priority than the unary minus, which may be surprising, since all other binary operators have lower priority than the unary minus.

This makes me question the usability of `x^y`. Quint already has `ipow` as a machine-readable alias for `x^y`. Hence, we could just drop `x^y` and use `x.ipow(y)` or `ipow(x, y)` instead.

**Comment by konnov** [comment]
Depends on #579
